lib/ui/painting.dart

```diff
@@ -4691,6 +4691,16 @@ class Canvas extends NativeFieldWrapperClass1 {
   @FfiNative<Void Function(Pointer<Void>)>('Canvas::restore', isLeaf: true)
   external void restore();
 
+  /// Pops the save stack until the count layer, if there is anything to pop.
+  /// Otherwise, does nothing.
```

Comment:
Add a comment about how this relates to getSaveCount(). Something along the lines of "the count layer that was previously retrieved with [getSaveCount]" or "After this method completes, the [count] should match the return value of [getSaveCount]".

Also, describe what happens if count > getSaveCount(). I believe it just gets ignored, but we should document that.

Reply:
How about:

```
/// Restores the save stack to a previous level as might be obtained from [getSaveCount].
/// If [count] is less than 1, the stack is restored to its initial state. If [count] is
/// greater than the current [getSaveCount] then nothing happens.
```

I hinted at what happens with a negative count and also implied that the initial save count is 1 (is that true?) - perhaps we should add a test that verifies these behaviors. I believe that one of the web implementations may not be protected against restoring too far which means that behavior will end up blowing up compared to the other implementations (check that restore protects against an empty stack and also that restoreToCount of values less than 1 all behave the same on all platforms).

Reply:
Get that, I have added test to cover those corner cases.